### DISCUSSION TITLE
"Propose changes" should select a public e-mail for commit, not the one that is set as "Primary"

### DISCUSSION THREAD

**Discussion by eliasdaler** [community/community#9214] [Pull Requests]
Hello.

I have several e-mails attached to my account. The "primary" one that I use for GitHub notifications and login info is private.
The public one is the one that I prefer to use for my commits.

However, when I use "Propose changes" feature in GitHub web UI, the first e-mail that gets selected in "Choose which email address to associate with this commit" is my primary e-mail, not my public e-mail.

I think that public e-mail should be selected by default for all fields like these so that people don't accidentally leak their private e-mail which can be set to "primary".

**Comment by sryze** [comment]
I ran into the same issue with primary email visibility.

Addionally, I've had a couple of occasions where someone else merged my PR to a public repo by squashing a the commits, and the final commit had my primary email attached instead of the public email that was used in my PR's commits. This is unexpected behavior IMO, and many other users may not not even realize that their account's email address has been leaked this way.

I think that GitHub should provide an option to simultaneously prevent usage of non-public emails in commits and to allow the public email as commit author if you have set one. Right now you can either allow all emails or none.
